contracts/modules/CompoundManager.sol

```diff
@@ -370,21 +376,27 @@ contract CompoundManager is OnlyOwnerModule {
      * @param _amount The amount of underlying token to redeem.
      */
     function redeemUnderlying(address _wallet, address _cToken, uint256 _amount) internal {
-        require(_cToken != address(0), "Compound: No market for target token");
-        require(_amount > 0, "Compound: amount cannot be 0");
+        require(_cToken != address(0), "CM: No market for target token");
+        require(_amount > 0, "CM: amount cannot be 0");
+        uint256 initialCTokenAmount = ERC20(_cToken).balanceOf(_wallet);
         invokeWallet(_wallet, _cToken, 0, abi.encodeWithSignature("redeemUnderlying(uint256)", _amount));
+        require(ERC20(_cToken).balanceOf(_wallet) < initialCTokenAmount, "CM: redeemUnderlying failed");
     }
 
     /**
      * @dev Borrows underlying tokens from a cToken contract.
      * @param _wallet The target wallet.
+     * @param _token The token contract.
      * @param _cToken The cToken contract.
      * @param _amount The amount of underlying tokens to borrow.
      */
-    function borrow(address _wallet, address _cToken, uint256 _amount) internal {
-        require(_cToken != address(0), "Compound: No market for target token");
-        require(_amount > 0, "Compound: amount cannot be 0");
+    function borrow(address _wallet, address _token, address _cToken, uint256 _amount) internal {
```

Comment:
Why do we need to pass the `_token`? We could get it from the `_cToken` contract and not have to add another parameter to the method. Or do the opposite and pass the token and get the cToken from the registry.

Reply:
We could but why pay the gas cost of an external call + extra read when we already have the data available? It's just an internal function.

Reply:
Of course, I did not notice it was an internal method.